core/domain/prod_validation_jobs_one_off.py

```diff
@@ -110,67 +351,3183 @@ def validate(cls, item):
+            'config_property_ids']
+
+        version = item.id[item.id.rfind('-') + 1:]
+        for (_, config_property_model) in config_property_model_tuples_list:
+            if int(config_property_model.version) < int(version):
+                cls.errors['config property model version check'].append((
+                    'Entity id %s: ConfigProperty model corresponding to '
+                    'id %s has a version %s which is less than the version %s '
+                    'in snapshot metadata model id' % (
+                        item.id, config_property_model.id,
+                        config_property_model.version, version)))
+
+    @classmethod
+    def _get_custom_validation_functions(cls):
+        return [cls._validate_config_property_model_version_from_item_id]
+
+
+class ConfigPropertySnapshotContentModelValidator(BaseModelValidator):
+    """Class for validating ConfigPropertySnapshotContentModel."""
+
+    @classmethod
+    def _get_model_id_regex(cls, item):
+        return '.'
+
+    @classmethod
+    def _get_json_properties_schema(cls, item):
+        return {}
+
+    @classmethod
+    def _get_model_domain_object_instances(cls, item):
+        return []
+
+    @classmethod
+    def _get_commit_cmd_domain_class(cls):
+        return None
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {
+            'config_property_ids': (
+                config_models.ConfigPropertyModel,
+                [item.id[:item.id.find('-')]]),
+        }
+
+    @classmethod
+    def _validate_config_property_model_version_from_item_id(cls, item):
+        """Validate that config property model corresponding to snapshot
+        content model has a version greater than or equal to the in item.id.
+
+        Args:
+            item: ConfigPropertySnapshotContentModel to validate.
+        """
+        _, config_property_model_tuples_list = cls.external_models[
+            'config_property_ids']
+
+        version = item.id[item.id.rfind('-') + 1:]
+        for (_, config_property_model) in config_property_model_tuples_list:
+            if int(config_property_model.version) < int(version):
+                cls.errors['config property model version check'].append((
+                    'Entity id %s: ConfigProperty model corresponding to '
+                    'id %s has a version %s which is less than the version %s '
+                    'in snapshot content model id' % (
+                        item.id, config_property_model.id,
+                        config_property_model.version, version)))
+
+    @classmethod
+    def _get_custom_validation_functions(cls):
+        return [cls._validate_config_property_model_version_from_item_id]
+
+
+class SentEmailModelValidator(BaseModelValidator):
+    """Class for validating SentEmailModels."""
+
+    @classmethod
+    def _get_model_id_regex(cls, item):
+        # Valid id: [intent].[random hash]
+        regex_string = '%s\\.\\..*$' % item.intent
```

Comment:
Try to have as many constraints as possible on the regex string, including '^' for "start of ID"  and more detailed specifications of what chars are allowed in the random hash (and how long it is). Be as strict as possible.

Reply:
Done.